# 3.2.3 The Mann-Whitney U test

- Perform a Mann-Whitney U test.
- Perform a Wilcoxon signed-rank test.
- Interpret the output of these tests.

Also known as the Wilcoxon rank-sum test, the Mann–Whitney U test is a non-parametric statistical test employed to compare the difference(s) between two independent groups.

Consider the following scenario: AB Consulting is conducting a survey among their employees to determine the effectiveness of two training programmes in improving employee productivity. They divide the participants into two groups of eight people. Participants have to write a test after completing the training programme. The higher the mark, the better the productivity of the participants. The following table indicates the test scores of participants out of 40:

|Group 1|Group 2|
|---|---|
12|18|
14|20|
15|21|
19|25|
22|28|
24|33|
29|36|
30|40|

The hypotheses are:
- $H_0$: There is no difference in median productivity between the two groups.
- $H_a$: There is a significant difference in median productivity between the two groups.



In [ ]:
# Import the necessary libraries.
import numpy as np
from scipy.stats import mannwhitneyu

In [ ]:
# Create sample data for the Mann–Whitney U test.
group1 = np.array([12, 14, 15, 19,
                   22, 24, 29, 30])
group2 = np.array([18, 20, 21, 25,
                   28, 33, 36, 40])

In [ ]:
# Perform the Mann–Whitney U test.
mannwhitney_stat, mannwhitney_p = mannwhitneyu(group1,
                                               group2)

# View the output.
print("Mann-Whitney statistic:", mannwhitney_stat)
print("Mann-Whitney p-value:", mannwhitney_p)

Mann-Whitney statistic: 17.0
Mann-Whitney p-value: 0.13038073038073036


The output indicates the test statistic ($U$) value is 17.0. The $p$-value is approximately 0.1304. Since the $p$-value is greater than the common alpha ($\alpha$) level of 0.05, we would not reject $H_0$, indicating that there is no significant difference between the two groups based on the provided data.

# Key information

The interpretation of the $p$-values would depend on the $\alpha$ level you set for your test (commonly 0.05 for a 5% significance level).​